Sources/DVR/Filter.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  Filter.swift
+//  DVR
+//
+//  Created by Jáir Myree on 6/18/21.
+//  Copyright © 2021 Venmo. All rights reserved.
+//
+
+import Foundation
+
+public struct Filter {
+    var headers : [String : String]
+    var queryParameters : [String : String]
+    var postDataParameters : [String : String]
+    var beforeRecordResponse : ((inout Foundation.URLResponse, inout Data?) -> ())
+    var beforeRecordRequest : ((inout URLRequest) -> ())
+    
+    public init(headers: [String] = [], queryParameters: [String] = [], postDataParameters: [String] = [], replacer: String = "Redacted", requestHook : @escaping ((inout URLRequest) -> ()) = {_ in } , responseHook : @escaping ((inout Foundation.URLResponse, inout Data?) -> ()) = {_,_  in } ) {
```

Comment:
Is `replacer` what is used in Python? To me it sounds like a function rather than a value. I would recommend `replacementString`. 

For the hooks, we should avoid using `inout` and made deliberate changes in AzureCore to avoid using it. It would be better to accept a URLRequest/Response and have the closure simply return a URLRequest/Response as well. This would most closely match the Python implementation.

It also would more closely match what we do in AzureCore regarding Pipeline stages:
https://github.com/Azure/azure-sdk-for-ios/blob/main/sdk/core/AzureCore/Source/Pipeline/PipelineStage.swift

Reply:
Sounds simple enough, I'll change it.